client/llb/source.go

```diff
@@ -455,7 +455,7 @@ func Differ(t DiffType, required bool) LocalOption {
 	})
 }
 
-func OCILayout(contentStoreID string, dig digest.Digest, opts ...OCILayoutOption) State {
+func OCILayout(ref string, opts ...OCILayoutOption) State {
```

Comment:
I don't fully get what we gain from going to a `ref string` vs handling the two separately. Aren't we better off having them separate, as we can require a hash? We never really added support for tag-based parsing of OCI layout source, mainly because there is no agreed standard for doing it. I have used the tags `org.opencontainers.image.ref.name` and `io.containerd.image.name`, as have many others, but there is no official agreement quite yet.

Reply:
Oh, OK, now I see it. You are adopting precisely that. No argument, very much in favour of it.